docs/content/preview/integrations/apache-spark/python-ysql.md

```diff
@@ -0,0 +1,292 @@
+    -----------+------------------+-----------+----------+---------
+     id        | integer          |           |          |
+     random    | double precision |           |          |
+     round_off | double precision |           |          |
+    ```
+
+    ```sql
+    ysql_pyspark=# SELECT COUNT(*) FROM test_copy;
+    ```
+
+    ```output
+     count
+    --------
+     100000
+    (1 row)
+    ```
+
+1. Use the `append` [SaveMode](https://spark.apache.org/docs/latest/sql-data-sources-load-save-functions.html#save-modes), to append data from `test_copy` to the `test` table as follows:
+
+    ```spark
+    >>> test_copy_Df = spark.read.jdbc(url=jdbcUrl, table="(select * from test_copy) test_copy_alias", properties=connectionProperties)
+    >>> test_copy_Df.createOrReplaceTempView("test_copy")
+    >>> spark.table("test_copy").write.mode("append").jdbc(url=jdbcUrl, table="test",     properties=connectionProperties)
+    ```
+
+1. Verify the changes using ysqlsh:
+
+    ```sql
+    ysql_pyspark=# SELECT COUNT(*) FROM test;
+    ```
+
+    ```output
+     count
+    --------
+     200000
+    (1 row)
+    ```
+
+## Parallelism
+
+To maintain parallelism while fetching the table content, create a DataFrame for the table `test` with some specific options as follows:
+
+- `numPartitions` - divides the whole task to `numPartitions` parallel tasks.
+- `lowerBound` - minimum value of the `partitionColumn` in a table.
+- `upperBound` - maximum value of the `partitionColumn` in a table.
+- `partitionColumn` - the column on the basis of which a partition occurs.
+
+These options help in breaking down the whole task into `numPartitions` parallel tasks on the basis of the `partitionColumn`, with the help of minimum and maximum value of the column.
+
+Additionally, the following two options help in optimizing the SQL queries executing on this dataframe if those SQL queries consist of some filters or aggregate functions by pushing down those filters and aggregates to the YugabyteDB through the JDBC connector.
+
+- `pushDownPredicate` - optimizes the query by pushing down the filters to YugabyteDB through the JDBC connector.
+- `pushDownAggregate` - optimizes the query by pushing down the aggregated to YugabyteDB through the JDBC connector.
+
+```spark
+>>> new_test_df = spark.read.jdbc(url=jdbcUrl, table="test", properties=connectionProperties,numPartitions=5, column="ceil", lowerBound=0, upperBound=20)
+
+>>> new_test_df.createOrReplaceTempView("test")
+>>> spark.sql("select sum(ceil) from test where id > 50000").show
+```
+
+```output
++---------+
+|sum(ceil)|
++---------+
+|1049414.0|
++---------+
+```
+
+### Verify parallelism
+
+To verify that the Spark job is created,
+
+1. Navigate to the Spark UI using <https://localhost:4040>. If your port 4040 is in use, then change the port to the one mentioned in the output of your [`pyspark`](#start-python-spark-shell-with-yugabytedb-driver) shell.
```

Comment:
the one mentioned when you started the [`pyspark`](#start-python-spark-shell-with-yugabytedb-driver) shell.